libr/core/cmd_open.c

```diff
@@ -1209,6 +1218,45 @@ static bool desc_list_json_cb(void *user, void *data, ut32 id) {
 	return true;
 }
 
+static bool cmd_op(RCore *core, char mode, int fd) {
+	int cur_fd = r_io_fd_get_current (core->io);
+	int next_fd = cur_fd;
+	switch (mode) {
+	case 0:
+		next_fd = fd;
+		break;
+	case 'n':
+		next_fd = r_io_fd_get_next (core->io, cur_fd);
+		break;
+	case 'p':
+		next_fd = r_io_fd_get_prev (core->io, cur_fd);
+		break;
+	case 'r':
+		next_fd = r_io_fd_get_next (core->io, cur_fd);
```

Comment:
judging from these few lines, it looks like you are assuming that `r_io_fd_get_next` does not rotate by itself, and I'm perfectly fine with this. However, I think you should change `r_id_storage_get_next`/`prev` to actually not rotate. The rotation can be easily done as you did, by checking for `-1` and getting the lowest (or highest, if you are rotating backward).

Reply:
i dont think next/prev should assume rotation at all. for me fds are a list, if idpool uses a ring internally thats out of the concept of FDs, and also, fd =0 should be valid, which doesnt seems to be valid in the current implementation

my current implementation does not rotate. just return false and give u the next/prev in case it rotates